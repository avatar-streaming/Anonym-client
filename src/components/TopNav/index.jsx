import React, { useState } from "react";
import { NavLink } from "react-router-dom";
import { useSelector } from "react-redux";
import useLogout from "../../hooks/useLogout";
import useSearch from "../../hooks/useSearch";

function TopNav() {
  const [searchTerm, setSearchTerm] = useState("");
  const handleSubmit = useSearch(searchTerm);
  const { _id: userId } = useSelector((state) => state.user.userInfo);
  const handleClick = useLogout();

  return (
    <nav className="nav-top">
      <div>
        <NavLink to="/" className="nav-top__title">Anonym</NavLink>
      </div>
      <div className="nav-top__search-bar">
        <form onSubmit={handleSubmit}>
          <input
            type="text"
            className="input-text"
            value={searchTerm}
            onChange={(e) => {
              if (e.target.value !== "\\") {
                setSearchTerm(e.target.value);
              }
            }}
          />
        </form>
      </div>
      <div>
        <NavLink to={`/streaming/${userId}`}>Streaming</NavLink>
        <NavLink to={`/user/${userId}`}>My Page</NavLink>
        <NavLink to="/auth/login" onClick={handleClick}>Logout</NavLink>
      </div>
    </nav>
  );
}

export default TopNav;
