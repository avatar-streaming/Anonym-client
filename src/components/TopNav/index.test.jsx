import React from "react";
import { Provider } from "react-redux";
import thunk from "redux-thunk";
import { Router } from "react-router";
import TopNav from ".";
import renderer from "react-test-renderer";
import configureMockStore from "redux-mock-store";
import { createMemoryHistory } from "history";

describe("<TopNav />", () => {
  const mockStore = configureMockStore([thunk]);
  const history = createMemoryHistory();
  let store = null;
  let component = null;

  beforeEach(() => {
    store = mockStore({
      user: {
        userInfo: {
          "_id": 0,
          userName: "user 0",
          thumnail: "thumnail 0",
          followers: [],
        },
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <Router history={history}>
          <TopNav />
        </Router>
      </Provider>
    );
  });

  it("should render with given state from redux store", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
