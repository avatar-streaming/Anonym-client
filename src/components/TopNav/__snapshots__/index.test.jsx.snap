// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNav /> should render with given state from redux store 1`] = `
<nav
  className="nav-top"
>
  <div>
    <a
      aria-current="page"
      className="nav-top__title active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      Anonym
    </a>
  </div>
  <div
    className="nav-top__search-bar"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        className="input-text"
        onChange={[Function]}
        type="text"
        value=""
      />
    </form>
  </div>
  <div>
    <a
      aria-current={null}
      href="/streaming/0"
      onClick={[Function]}
    >
      Streaming
    </a>
    <a
      aria-current={null}
      href="/user/0"
      onClick={[Function]}
    >
      My Page
    </a>
    <a
      aria-current={null}
      href="/auth/login"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</nav>
`;
